# Pauli Matrices

파울리 행렬의 아이겐밸류와 아이겐벡터를 구하시오.  

$$ s_x = \begin{pmatrix} 0 & 1 \\ 1 & 0 \end{pmatrix}, \quad
s_y = \begin{pmatrix} 0 & -j \\ j & 0 \end{pmatrix}, \quad
s_z  = \begin{pmatrix} 1 & 0 \\ 0 & -1 \end{pmatrix}. $$

In [5]:
# 파이썬에서 허수 i를 j 혹은 J로 표현한다. 
# 예를 들어,
a = 3 + 2j
b = 5 - 3j

In [6]:
import numpy as np

s_x = np.array([[0, 1], [1, 0]])
s_y = np.array([[0, -1j], [1j, 0]])
s_z = np.array([[1, 0], [0, -1]])

In [27]:
# 넘파이의 선형대수 관련 연산 중 아이겐밸류, 아이겐벡터를 바로 구해주는 eig 메소드를 사용
np.linalg.eig(s_x)

(array([ 1., -1.]),
 array([[ 0.70710678, -0.70710678],
        [ 0.70710678,  0.70710678]]))

In [28]:
np.linalg.eig(s_y)

(array([ 1.+0.j, -1.+0.j]),
 array([[-0.        -0.70710678j,  0.70710678+0.j        ],
        [ 0.70710678+0.j        ,  0.        -0.70710678j]]))

In [29]:
np.linalg.eig(s_z)

(array([ 1., -1.]),
 array([[1., 0.],
        [0., 1.]]))

# Normal Modes 

(고전역학 by Thornton & Marion 챕터 12.2 및 예제 12.3 참조)
용수철 상수가 $k$인 용수철에 연결된 두 개의 물체의 노말 모드를 구해보자. 두 개의 물체의 운동방정식은 다음으로 주어진다.
$$ -m w^2 x_1 = -2kx_1 +k x_2, \\
-m w^2 x_2 = k x_1 - 2 k x_2.$$
eigenfrequencies $w$는 다음의 행렬식을 통해 구할 수 있다.
$$ \frac{k}{m w^2} \begin{pmatrix} 2 & -1 \\ -1 & 2 \end{pmatrix} 
\begin{pmatrix} x_1 \\ x_2 \end{pmatrix}=
\lambda \begin{pmatrix} x_1 \\ x_2 \end{pmatrix} $$

In [1]:
# 위의 행렬의 고유값을 구하자.

(array([3., 1.]), array([[ 0.70710678,  0.70710678],
        [-0.70710678,  0.70710678]]))

In [30]:
matrix = np.array([[2, -1], [-1, 2]])
np.linalg.eig(matrix)

(array([3., 1.]),
 array([[ 0.70710678,  0.70710678],
        [-0.70710678,  0.70710678]]))

따라서 eigenfrequencies는
$$ w_1 = \sqrt{\frac{k}{m}}, \quad w_2 = \sqrt{\frac{3k}{m}}. $$

위에서 구한 아이겐벡터와 아이겐밸류의 의미를 설명하여라.

운동방정식을 행렬로 정리하면 다음과 같다.

$$ \frac{k}{m w^2} \begin{pmatrix} 2 & -1 \\ -1 & 2 \end{pmatrix} 
\begin{pmatrix} x_1 \\ x_2 \end{pmatrix}=
\begin{pmatrix} x_1 \\ x_2 \end{pmatrix} $$

여기서 

$$ \frac{k}{m w^2} $$

를 우변으로 옮기면 식은 다음과 같이 정리된다.

$$ \begin{pmatrix} 2 & -1 \\ -1 & 2 \end{pmatrix} 
\begin{pmatrix} x_1 \\ x_2 \end{pmatrix}=
\frac{m w^2}{k} \begin{pmatrix} x_1 \\ x_2 \end{pmatrix} $$

즉 아이겐벡터와 아이겐밸류를 구할 시 

$$ \frac{m w^2}{k} $$

가 아이겐밸류가 된다.

위에서 아이겐밸류를 구하면 1, 3이 나온다.

아이겐밸류가 3일 때에는 아이겐벡터가 [0.70710678, -0.70710678]이므로 두 개의 진자의 진폭의 크기는 같지만 방향이 다르고 아이겐밸류가 1일 때는 [0.70710678, 0.70710678]이므로 두 개의 진자의 진폭의 크기도 같고 방향도 같다.